framework/src/meshmodifiers/BreakMeshByBlock.C

```diff
@@ -0,0 +1,170 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BreakMeshByBlock.h"
+#include "MooseMesh.h"
+
+registerMooseObject("MooseApp", BreakMeshByBlock);
+
+template <>
+InputParameters
+validParams<BreakMeshByBlock>()
+{
+  InputParameters params = validParams<BreakMeshByBlockBase>();
+  params.addClassDescription("Break boundaries based on the subdomains to which their sides are "
+                             "attached. Naming convention for the new boundaries will be the old "
+                             "boundary name plus \"_to_\" plus the subdomain name. At the moment"
+                             "this only works on REPLICATED mesh");
+  params.addParam<std::vector<BoundaryName>>(
+      "boundaries", "Boundaries to be broken. Default means to break all boundaries");
+  return params;
+}
+
+BreakMeshByBlock::BreakMeshByBlock(const InputParameters & parameters)
+  : BreakMeshByBlockBase(parameters)
+{
+}
+
+void
+BreakMeshByBlock::modify()
+{
+  // TODO remove when distributed MESH capabilities are implemented
+  _mesh_ptr->errorIfDistributedMesh("BreakMeshByBlock only works on a REPLICATED mesh");
+  checkInputParameter();
+
+  // initialize the node to elemen map
+  const std::map<dof_id_type, std::vector<dof_id_type>> & node_to_elem_map =
+      _mesh_ptr->nodeToElemMap();
+
+  // keep track of all paired subdomains
+  std::set<std::vector<SubdomainID>> block_pairs;
+
+  for (auto node_it = node_to_elem_map.begin(); node_it != node_to_elem_map.end(); ++node_it)
+  {
+    const dof_id_type current_node_id = node_it->first;
+    const Node current_node = _mesh_ptr->getMesh().node(current_node_id);
```

Comment:
Please save off a `libMesh::Mesh &` (what comes back from `getMesh()`) so you're not calling that all over the place.